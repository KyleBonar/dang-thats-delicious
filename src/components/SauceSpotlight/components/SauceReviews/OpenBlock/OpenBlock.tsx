import RatingBlock from "../components/RatingBlock/RatingBlock";
import * as React from "react";
import { IReview } from "../../../../../redux/reviews/types";

export interface IOpenBlock {
  review: IReview;
}

const OpenBlock: React.FC<IOpenBlock> = props => {
  const { review } = props;

  return (
    <div>
      {/* Overall */}
      {review.overall && (
        <RatingBlock
          name="Overall"
          txt={review.overall.txt}
          rating={review.overall.rating}
        />
      )}
      {/* Aroma */}
      {review.aroma && (review.aroma.rating > 0 || review.aroma.txt) && (
        <RatingBlock
          name="Aroma"
          txt={review.aroma.txt}
          rating={review.aroma.rating}
        />
      )}
      {/* Taste */}
      {review.taste && (review.taste.rating > 0 || review.taste.txt) && (
        <RatingBlock
          name="Taste"
          txt={review.taste.txt}
          rating={review.taste.rating}
        />
      )}
      {/* Label */}
      {review.label && (review.label.rating > 0 || review.label.txt) && (
        <RatingBlock
          name="Label"
          txt={review.label.txt}
          rating={review.label.rating}
        />
      )}
      {/* Heat */}
      {review.heat && (review.heat.rating > 0 || review.heat.txt) && (
        <RatingBlock
          name="Heat"
          txt={review.heat.txt}
          rating={review.heat.rating}
        />
      )}
      {/* Note */}
      {review.note && review.note.txt && (
        <RatingBlock name="Note" txt={review.note.txt} />
      )}
    </div>
  );
};

export default OpenBlock;
