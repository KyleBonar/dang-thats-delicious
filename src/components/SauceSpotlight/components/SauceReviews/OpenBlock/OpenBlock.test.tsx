import * as React from "react";
import * as enzyme from "enzyme";

import OpenBlock, { IOpenBlock } from "./OpenBlock";

import {
  fakeReview,
  ITERATION_SIZE
} from "../../../../../utils/testUtils/testUtils";

const fakeSauceReviewBlocks = (): IOpenBlock => {
  return {
    review: fakeReview()
  };
};

describe("<OpenBlock />", () => {
  let wrappers: any = [];
  let props: IOpenBlock[] = [];

  beforeAll(() => {
    wrappers = new Array(ITERATION_SIZE).fill(null).map(() => {
      // Generate props
      const fakeProp = fakeSauceReviewBlocks();

      // Add props to collector
      props.push(fakeProp);

      return enzyme.shallow(<OpenBlock {...fakeProp} />);
    });
  });

  it("renders", () => {
    wrappers.forEach(wrapper => {
      expect(wrapper.exists()).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    wrappers.forEach(wrapper => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("renders a RatingBlock component for Aroma if Aroma is valid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab aroma
      const {
        review: { aroma }
      } = props[ind];

      // Make sure aroma is legit
      if (!aroma) return;
      if (aroma.rating === 0 && (!aroma.txt || aroma.txt.length === 0)) return;

      // Find component
      expect(wrapper.find("RatingBlock[name='Aroma']").exists()).toBeTruthy();
    });
  });

  it("does not render a RatingBlock component for Aroma if Aroma is invalid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab aroma
      const {
        review: { aroma }
      } = props[ind];

      // Make sure aroma is invalid
      if (aroma && (aroma?.rating > 0 || aroma?.txt?.length > 0)) return;

      // Look for component -- should not find it
      expect(wrapper.find("RatingBlock[name='Aroma']").exists()).toBeFalsy();
    });
  });

  it("renders a RatingBlock component for Taste if Taste is valid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab taste
      const {
        review: { taste }
      } = props[ind];

      // Make sure taste is legit
      if (!taste) return;
      if (taste.rating === 0 && (!taste.txt || taste.txt.length === 0)) return;

      // Find component
      expect(wrapper.find("RatingBlock[name='Taste']").exists()).toBeTruthy();
    });
  });

  it("does not render a RatingBlock component for Taste if Taste is invalid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab taste
      const {
        review: { taste }
      } = props[ind];

      // Make sure taste is invalid
      if (taste && (taste?.rating > 0 || taste?.txt?.length > 0)) return;

      // Look for component -- should not find it
      expect(wrapper.find("RatingBlock[name='Taste']").exists()).toBeFalsy();
    });
  });

  it("renders a RatingBlock component for Label if Label is valid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab label
      const {
        review: { label }
      } = props[ind];

      // Make sure label is legit
      if (!label) return;
      if (label.rating === 0 && (!label.txt || label.txt.length === 0)) return;

      // Find component
      expect(wrapper.find("RatingBlock[name='Label']").exists()).toBeTruthy();
    });
  });

  it("renders a RatingBlock component for Heat if Heat is valid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab heat
      const {
        review: { heat }
      } = props[ind];

      // Make sure heat is legit
      if (!heat) return;
      if (heat.rating === 0 && (!heat.txt || heat.txt.length === 0)) return;

      // Find component
      expect(wrapper.find("RatingBlock[name='Heat']").exists()).toBeTruthy();
    });
  });

  it("renders a RatingBlock component for Note if Note is valid", () => {
    wrappers.forEach((wrapper, ind) => {
      // Grab note
      const {
        review: { note }
      } = props[ind];

      // Make sure note is legit
      if (!note) return;
      if (!note.txt) return;
      if (note.txt.length === 0) return;

      // Find component
      expect(wrapper.find("RatingBlock[name='Note']").exists()).toBeTruthy();
    });
  });
});
