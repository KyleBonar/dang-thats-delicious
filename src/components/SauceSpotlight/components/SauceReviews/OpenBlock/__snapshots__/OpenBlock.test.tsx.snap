// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpenBlock /> matches snapshot 1`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={0}
  />
  <RatingBlock
    name="Taste"
    rating={3}
    txt="Delectus vel ipsa tempora."
  />
  <RatingBlock
    name="Label"
    rating={3}
  />
  <RatingBlock
    name="Heat"
    rating={0}
    txt="Illum animi vel provident dicta perferendis dolorum."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 2`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={0}
  />
  <RatingBlock
    name="Aroma"
    rating={1}
    txt="Quam nulla est unde sit ex."
  />
  <RatingBlock
    name="Taste"
    rating={0}
    txt="Eaque necessitatibus sit suscipit."
  />
  <RatingBlock
    name="Label"
    rating={2}
    txt="Autem sit aspernatur aspernatur quam eligendi."
  />
  <RatingBlock
    name="Heat"
    rating={5}
    txt="Quae corporis qui aut nobis."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 3`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
    txt="Qui voluptatem cumque nihil iste."
  />
  <RatingBlock
    name="Aroma"
    rating={3}
  />
  <RatingBlock
    name="Taste"
    rating={1}
  />
  <RatingBlock
    name="Heat"
    rating={4}
    txt="Quos eum voluptate sit saepe assumenda deserunt."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 4`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
    txt="Esse quos minus deserunt nihil quasi minus quo."
  />
  <RatingBlock
    name="Aroma"
    rating={3}
    txt="Provident eaque atque quos."
  />
  <RatingBlock
    name="Taste"
    rating={3}
  />
  <RatingBlock
    name="Heat"
    rating={1}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 5`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
  />
  <RatingBlock
    name="Label"
    rating={1}
    txt="Quia aut sint cumque sunt."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 6`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={1}
    txt="Voluptas eos quam."
  />
  <RatingBlock
    name="Aroma"
    rating={4}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 7`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={3}
    txt="Reiciendis ut eos commodi at impedit sint perferendis in dolorem."
  />
  <RatingBlock
    name="Taste"
    rating={4}
    txt="Omnis quo aut quia cumque architecto eos quos."
  />
  <RatingBlock
    name="Label"
    rating={4}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 8`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={3}
    txt="Impedit dolores saepe dolor incidunt vero quia voluptatibus consequatur."
  />
  <RatingBlock
    name="Label"
    rating={2}
    txt="Qui qui ab officiis omnis rem officiis et placeat ipsum."
  />
  <RatingBlock
    name="Heat"
    rating={2}
    txt="Et voluptas repellendus nisi corrupti quas eligendi."
  />
  <RatingBlock
    name="Note"
    txt="Et ea quia sunt est aut et sunt libero."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 9`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
    txt="Nulla sit officia doloremque non sit a perferendis soluta dignissimos."
  />
  <RatingBlock
    name="Label"
    rating={5}
    txt="Sed consequatur ut sapiente molestiae dolores."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 10`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
    txt="Ipsam fuga rem eius est nihil sit enim itaque nostrum."
  />
  <RatingBlock
    name="Aroma"
    rating={3}
  />
  <RatingBlock
    name="Label"
    rating={3}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 11`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
    txt="Eaque quis omnis ut praesentium vel."
  />
  <RatingBlock
    name="Taste"
    rating={3}
  />
  <RatingBlock
    name="Label"
    rating={4}
  />
  <RatingBlock
    name="Note"
    txt="Eum doloribus et perspiciatis sunt."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 12`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={1}
  />
  <RatingBlock
    name="Taste"
    rating={2}
  />
  <RatingBlock
    name="Label"
    rating={1}
  />
  <RatingBlock
    name="Note"
    txt="Suscipit reiciendis in exercitationem et error tempore aspernatur soluta."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 13`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={3}
  />
  <RatingBlock
    name="Aroma"
    rating={2}
    txt="Dolores voluptas quidem excepturi temporibus."
  />
  <RatingBlock
    name="Taste"
    rating={2}
    txt="Ratione saepe id officia."
  />
  <RatingBlock
    name="Heat"
    rating={5}
  />
  <RatingBlock
    name="Note"
    txt="Eligendi excepturi fugit beatae ut ea."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 14`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={1}
    txt="Excepturi laborum veniam asperiores et consectetur animi."
  />
  <RatingBlock
    name="Aroma"
    rating={3}
    txt="Nam doloribus et facere."
  />
  <RatingBlock
    name="Heat"
    rating={5}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 15`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
  />
  <RatingBlock
    name="Aroma"
    rating={0}
    txt="Culpa cum architecto nulla ut molestias nisi corporis nulla."
  />
  <RatingBlock
    name="Taste"
    rating={1}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 16`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={5}
  />
  <RatingBlock
    name="Aroma"
    rating={2}
  />
  <RatingBlock
    name="Taste"
    rating={3}
  />
  <RatingBlock
    name="Heat"
    rating={4}
  />
  <RatingBlock
    name="Note"
    txt="Quae ut dolores quam ea dolores."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 17`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={3}
  />
  <RatingBlock
    name="Taste"
    rating={1}
    txt="Voluptas delectus iure veniam est consequatur."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 18`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
  />
  <RatingBlock
    name="Aroma"
    rating={5}
    txt="Nulla maiores pariatur amet harum maxime a."
  />
  <RatingBlock
    name="Taste"
    rating={4}
  />
  <RatingBlock
    name="Label"
    rating={4}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 19`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
  />
  <RatingBlock
    name="Label"
    rating={2}
    txt="Odio id id repellat maiores iure fuga sit."
  />
  <RatingBlock
    name="Heat"
    rating={1}
    txt="Exercitationem excepturi eligendi facilis similique optio hic perspiciatis dolor."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 20`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
    txt="Ut nobis velit velit perspiciatis mollitia ut corrupti sunt."
  />
  <RatingBlock
    name="Aroma"
    rating={4}
    txt="Aut dolore labore vitae aperiam voluptas impedit."
  />
  <RatingBlock
    name="Taste"
    rating={4}
  />
  <RatingBlock
    name="Label"
    rating={4}
    txt="Porro excepturi aperiam qui voluptatem rerum."
  />
  <RatingBlock
    name="Heat"
    rating={0}
    txt="Mollitia quis dolor non id cum enim mollitia voluptas accusamus."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 21`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={1}
  />
  <RatingBlock
    name="Label"
    rating={4}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 22`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
  />
  <RatingBlock
    name="Label"
    rating={1}
    txt="Vitae magni a corrupti eveniet."
  />
  <RatingBlock
    name="Heat"
    rating={1}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 23`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={0}
    txt="Vel tenetur expedita veritatis."
  />
  <RatingBlock
    name="Aroma"
    rating={2}
  />
  <RatingBlock
    name="Taste"
    rating={1}
  />
  <RatingBlock
    name="Heat"
    rating={5}
    txt="Ducimus sunt sed totam corporis error quae laudantium quos sequi."
  />
  <RatingBlock
    name="Note"
    txt="Exercitationem nostrum voluptas quia nam quisquam quia nostrum rerum."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 24`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={2}
    txt="Dolores quasi iure modi."
  />
  <RatingBlock
    name="Aroma"
    rating={3}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 25`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={3}
    txt="Molestias perferendis doloremque qui voluptatibus."
  />
  <RatingBlock
    name="Taste"
    rating={5}
  />
  <RatingBlock
    name="Note"
    txt="Eum blanditiis ut nisi et et vel quasi."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 26`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={3}
    txt="Ipsam id nisi explicabo accusantium non id aliquid tempora praesentium."
  />
  <RatingBlock
    name="Label"
    rating={1}
  />
  <RatingBlock
    name="Heat"
    rating={3}
    txt="Iure velit officia in excepturi."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 27`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={5}
    txt="Veritatis in nisi quia et ea velit dolorem libero."
  />
  <RatingBlock
    name="Taste"
    rating={3}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 28`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
  />
  <RatingBlock
    name="Aroma"
    rating={3}
    txt="Dolorem non earum et."
  />
  <RatingBlock
    name="Label"
    rating={3}
    txt="Esse inventore enim necessitatibus nihil autem assumenda consequatur."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 29`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={0}
  />
  <RatingBlock
    name="Heat"
    rating={1}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 30`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={4}
    txt="Dolore quidem sunt et at earum molestiae quae."
  />
  <RatingBlock
    name="Taste"
    rating={4}
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 31`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={1}
  />
  <RatingBlock
    name="Aroma"
    rating={2}
    txt="Blanditiis omnis amet ipsa consectetur."
  />
  <RatingBlock
    name="Label"
    rating={1}
  />
  <RatingBlock
    name="Heat"
    rating={4}
    txt="Aliquam est consequuntur."
  />
</div>
`;

exports[`<OpenBlock /> matches snapshot 32`] = `
<div>
  <RatingBlock
    name="Overall"
    rating={0}
    txt="Fuga placeat odio est quos voluptates in."
  />
  <RatingBlock
    name="Aroma"
    rating={3}
  />
  <RatingBlock
    name="Taste"
    rating={1}
  />
  <RatingBlock
    name="Heat"
    rating={4}
  />
</div>
`;
